packages/client/src/utils/buttonActions.js

```diff
@@ -531,6 +532,19 @@ export const enrichButtonActions = (actions, context) => {
           const result = await callback()
           if (result === false) {
             return
+          }
+          if (result?.valid === false) {
+            await scrollHandler(
+              {
+                [`##eventHandlerType`]: "Scroll To Field",
+                parameters: {
+                  block: "start",
+                  componentId: action.parameters.componentId,
+                  field: result.fields?.find(field => !field.valid)?.field,
+                },
+              },
+              context
+            )
```

Comment:
As mentioned above, we can remove this if we just handle scrolling to the first invalid field as part of the formAPI validate function.